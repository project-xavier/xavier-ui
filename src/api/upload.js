import ApiClient from './apiClient';

export function uploadFile(customerId, formData, config = {}) {
    return ApiClient.post(`/upload/${customerId}`, formData, config);
}
