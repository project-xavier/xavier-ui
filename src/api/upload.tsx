import ApiClient from './apiClient';
import { AxiosPromise } from 'axios';

export function uploadFile(formData: FormData, config = {}): AxiosPromise {
  return ApiClient.post('/upload', formData, config);
}
