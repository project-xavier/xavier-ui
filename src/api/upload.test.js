jest.mock('./apiClient', () => ({
    get: jest.fn(() => Promise.resolve({})),
    post: jest.fn(() => Promise.resolve({})),
    put: jest.fn(() => Promise.resolve({})),
    delete: jest.fn(() => Promise.resolve({}))
}));

import * as actionGenerator from './upload';

describe('uploadFile', () => {
    it('returns a state object', () => {
        const customerId = 'myCustomerId';
        const file = new File([ '' ], 'myFile.zip');
        const config = {};
        expect(actionGenerator.uploadFile(customerId, file, config)).toMatchSnapshot();
    });
});
