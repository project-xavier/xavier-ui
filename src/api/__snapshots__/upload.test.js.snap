// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uploadFile returns a state object 1`] = `Promise {}`;
